design/authenticators/authn_gcp/authn_gcp_solution_design.md

```diff
@@ -0,0 +1,312 @@
+
+| **Term** | **Description** |
+|----------|-----------------|
+|  GCP| Google Cloud Platform                 |
+|  Google compute engine (GCE)| Virtual machine in google (similar to AWS EC2)                  |
+|  Environment-provided service account| An identity for accessing private data on behalf of a service account inside Google Cloud environments |
+|  Service account key| App can self sign a google token with the service account key in order to accessing private data on behalf of a service account outside Google Cloud environments|
+
+
+
+## Useful links
+- [Feature doc](https://ca-il-confluence.il.cyber-ark.com/display/rndp/Conjur+GCP+Authenticator#/)
+- [GCP authentication overview]( https://cloud.google.com/docs/authentication)
+- [Google metadata server](https://cloud.google.com/compute/docs/storing-retrieving-metadata#querying)
+
+## Issue description
+We want enable GCE hosts to authenticate Conjur with their google identity in order to retrieve secrets from Conjur.
+
+## Out of scope
+The only authentication strategy we are supporting is Environment-provided service account in google Compute engine.
+Other strategies are not supported: Service account key, OAuth 2.0 client, API key
+Other google components are not supported:  App Engine, GKE, Cloud Run, Cloud Functions
+
+## Solution
+We will add a new GCP authenticator to Conjur, which GCE hosts will can authenticate with by delivering their host Conjur identity and google identity as JWT token.
+
+As mentioned in the [feature doc](https://ca-il-confluence.il.cyber-ark.com/display/rndp/Conjur+GCP+Authenticator#/), before any authentication 
+request is sent to Conjur, the admin will load the authenticator policy: 
+
+```yaml
+# policy id needs to match the convention `conjur/authn-azure/<service ID>`
+- !policy
+  id: conjur/authn-gcp/mygcp
+  body:
+  - !webservice
+      
+  - !group apps
+  
+  - !permit
+    role: !group apps
+    privilege: [ read, authenticate ]
+    resource: !webservice
+```
+
+All the following configurations will have constants values and will not require any action from the customer:
+
+* **provider-uri** = `https://www.googleapis.com/oauth2/v1/certs` #represnts the address to fetch the public key in order to validate the JWT token
+* **mandatory-claims** = `iss, exp, aud, iat, nbf` #the list of the mandatory claims to validate in the JWT token
+* **iss** = `https://accounts.google.com` #the issuer claim we should validate in the JWT token
+* **aud** = `conjur` #the audience claim we should validate in the JWT token
+
+### GCP Resource Restrictions
+To authenticate with Conjur, GCP-specific fields will need to be provided in the host annotations of the Conjur host identity with the following options:
+
+* **project_id** - A customizable unique identifier for your project.
+* **instance_name** - The name of the GCE machine
+* **service_account_id** - The google instance identity   
+* **service_account_email** - The email of the instance identity
+
```

Comment:
what are validation of above fields ?

Reply:
The fileds should match the host annotaitons.

Reply:
@eranha above you gave strict instructions to validate project id maybe there are for other fields?

Reply:
@inbal it was in response to Ofira's question.